packages/jsonmodels/ledgerstate.go

```diff
@@ -484,32 +484,20 @@ type OutputMetadata struct {
 	Solid               bool                `json:"solid"`
 	SolidificationTime  int64               `json:"solidificationTime"`
 	ConsumerCount       int                 `json:"consumerCount"`
-	FirstConsumer       string              `json:"firstConsumer,omitempty"`
 	ConfirmedConsumer   string              `json:"confirmedConsumer,omitempty"`
 	GradeOfFinality     gof.GradeOfFinality `json:"gradeOfFinality"`
 	GradeOfFinalityTime int64               `json:"gradeOfFinalityTime"`
 }
 
 // NewOutputMetadata returns the OutputMetadata from the given ledgerstate.OutputMetadata.
-func NewOutputMetadata(outputMetadata *ledgerstate.OutputMetadata) *OutputMetadata {
-	firstConsumer := ""
-	// omit firstConsumer field if it hasn't been consumed yet
-	if outputMetadata.FirstConsumer().Base58() != ledgerstate.GenesisTransactionID.Base58() {
-		firstConsumer = outputMetadata.FirstConsumer().Base58()
-	}
-	confirmedConsumer := ""
-	// omit confirmedConsumer field if it hasn't been consumed yet by a confirmed transaction
-	if outputMetadata.ConfirmedConsumer().Base58() != ledgerstate.GenesisTransactionID.Base58() {
-		confirmedConsumer = outputMetadata.ConfirmedConsumer().Base58()
-	}
+func NewOutputMetadata(outputMetadata *ledgerstate.OutputMetadata, confirmedConsumer *ledgerstate.Consumer) *OutputMetadata {
```

Comment:
I don't think this is practical at all. The consumer object contains redundant information as this is the output metadata. It should be sufficient to just pass the `transactionID` and take the base58 representation for this json model.